UnityProject/Assets/Scripts/Objects/Other/SecBarrier.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Objects.Security
+{
+	[RequireComponent(typeof(AccessRestrictions))]
+	[RequireComponent(typeof(Construction.WrenchSecurable))]
+	public class SecBarrier : MonoBehaviour, IPredictedCheckedInteractable<HandApply>
+	{
+		private AccessRestrictions accessRestrictions;
+		public AccessRestrictions AccessRestrictions
+		{
+			get
+			{
+				if (accessRestrictions == false)
+				{
+					accessRestrictions = GetComponent<AccessRestrictions>();
+				}
+				return accessRestrictions;
+			}
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			//start with the default HandApply WillInteract logic.
+			if (!DefaultWillInteract.Default(interaction, side)) return false;
+
+			//Only go through with interaction if correct access restriction
+			if (AccessRestrictions.CheckAccess(interaction.Performer))
+			{
+				//Player needs wrench to interact
+				if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Wrench))
+				{
+					return false;
+				}
+			}
+
+			return true;
+		}
+
+		public void ClientPredictInteraction(HandApply interaction)
+		{
+
+		}
+
+
+		public void ServerRollbackClient(HandApply interaction)
+		{
+
+		}
+
+		//invoked when the server recieves the interaction request and WIllinteract returns true
+		public async void ServerPerformInteraction(HandApply interaction)
+		{
+			if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Wrench))
+			{
+				//Notify player that they are unable to wrench down barrier
+				Chat.AddActionMsgToChat(interaction, "You try to wrench down the barrier...",
+					"The barrier denies you access.");
+			}
+			else {
+				Chat.AddActionMsgToChat(interaction, "ACCESS DENIED","");
```

Comment:
perhaps this message should be editable, as the script is not necessarily always going to be used for sec barriers.

Reply:
Gotchya, in that case should I generalize the script for wrenchable objects that have access restrictions? I assumed I was just making a sec barrier only script

Reply:
> perhaps this message should be editable, as the script is not necessarily always going to be used for sec barriers.

the conditions of ``WillInteract`` might be way too specific to be used for anything else, but still parametrising the string with a serialized field is always good.

Reply:
> Gotchya, in that case should I generalize the script for wrenchable objects that have access restrictions? I assumed I was just making a sec barrier only script

generalized scripts are * always * preferred I would say, though I'm not an arbiter for this stuff.

Reply:
Just renamed and generalized it further!